src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,133 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static final RegistrationService REGISTRATION = new RegistrationServiceImpl();
+    private static final String EMPTY = "";
+    private static final int CORRECT_AGE = 18;
+    private static final int INCORRECT_AGE = 16;
+    private static final String INCORRECT_LOGIN_WITH_WHITESPACE = "Log in";
+    private static final String CORRECT_PASSWORD = "passWord6";
+    private static final String INCORRECT_PASSWORD_SHORT = "short";
+    private static final String INCORRECT_PASSWORD_WITH_WHITESPACE = "pass Word6";
+    private static final String INCORRECT_PASSWORD_WITHOUT_DIGIT = "passWord";
+    private static final String INCORRECT_PASSWORD_WITHOUT_UPPERCASE = "password6";
```

Comment:
just use simple values without creating constants for everything, it's okay in tests
having constant just for minimal required age would be nice, because you could test like MIN_AGE - 1 and MIN_AGE + 1
the rest is redundant